appveyor.yml

```diff
@@ -1,71 +1,84 @@
 version: '{build}'
 
-os: Visual Studio 2015
-
 environment:
   matrix:
-  - Toolset: v140
-  - Toolset: v120
-  - Toolset: v110
-  - Toolset: v100
+    - compiler: msvc-10-seh
+      generator: "Visual Studio 10 2010"
+
+    - compiler: msvc-11-seh
+      generator: "Visual Studio 11 2012"
+
+    - compiler: msvc-11-seh
+      generator: "Visual Studio 11 2012 Win64"
+
+    - compiler: msvc-12-seh
+      generator: "Visual Studio 12 2013"
+
+    - compiler: msvc-12-seh
+      generator: "Visual Studio 12 2013 Win64"
 
-platform:
-  - Win32
-  - x64
+    - compiler: msvc-14-seh
+      generator: "Visual Studio 14 2015"
+
+    - compiler: msvc-14-seh
+      generator: "Visual Studio 14 2015 Win64"
+
+    - compiler: gcc-5.3.0-posix
+      generator: "MinGW Makefiles"
 
 configuration:
-#  - Release
   - Debug
+  - Release
 
 build:
   verbosity: minimal
 
-artifacts:
-  - path: '_build/Testing/Temporary/*'
-    name: test_results
-
-before_build:
+install:
 - ps: |
-    Write-Output "Configuration: $env:CONFIGURATION"
-    Write-Output "Platform: $env:PLATFORM"
-    $generator = switch ($env:TOOLSET)
-    {
-        "v140" {"Visual Studio 14 2015"}
-        "v120" {"Visual Studio 12 2013"}
-        "v110" {"Visual Studio 11 2012"}
-        "v100" {"Visual Studio 10 2010"}
-    }
-    if ($env:PLATFORM -eq "x64")
-    {
-        $generator = "$generator Win64"
+    Write-Output "Compiler: $env:compiler"
+    Write-Output "Generator: $env:generator"
+
+    # git bash conflicts with MinGW makefiles
+    if ($env:generator -eq "MinGW Makefiles") {
+        $env:path = $env:path.replace("C:\Program Files\Git\usr\bin;", "")
+        if ($env:compiler -eq "gcc-5.3.0-posix") {
+            $cxx_path = "C:\mingw-w64\i686-5.3.0-posix-dwarf-rt_v4-rev0\mingw32\bin"
```

Comment:
Sorry for bugging you again what with PR #723 and #856? I'm sorry if I'm too intrusive :blush: